modules/java/CMakeLists.txt

```diff
@@ -84,42 +66,34 @@ endif()
 
   include(UseJava)
 
+if ("${Java_VERSION}" VERSION_LESS 1.8)
   add_jar(syslog-ng-core
       SOURCES ${JAVA_SOURCES}
       OUTPUT_DIR ${CMAKE_CURRENT_BINARY_DIR}
   )
 
-  gen_java_header(
-      ${CMAKE_CURRENT_BINARY_DIR}/org_syslog_ng_LogMessage.h
-      ${CMAKE_CURRENT_BINARY_DIR}/syslog-ng-core.jar
-      ${CMAKE_CURRENT_BINARY_DIR}
-      "org.syslog_ng.LogMessage"
-      "syslog-ng-core"
+  create_javah(GENERATED_FILES BUILT_HEADERS
+        CLASSES    org.syslog_ng.LogMessage org.syslog_ng.InternalMessageSender org.syslog_ng.LogDestination org.syslog_ng.LogTemplate
+        CLASSPATH ${CMAKE_CURRENT_SOURCE_DIR}/src/main/java/
```

Comment:
This seems to be tricky. Why did you set CLASSPATH to the source dir?

Reply:
Long story short, it works if you specify the source instead of classpath, but that's as you pointed out not a good thing to do. Now corrected with a proper `get_property` call.